rerun_py/rerun_sdk/rerun/error_utils.py

```diff
@@ -1,15 +1,64 @@
 from __future__ import annotations
 
+import functools
 import inspect
-import logging
+import threading
+import warnings
+from types import TracebackType
+from typing import Any, Callable, TypeVar, cast
 
-import rerun
-from rerun.recording_stream import RecordingStream
+from .recording_stream import RecordingStream
 
 __all__ = [
     "_send_warning",
 ]
 
+_TFunc = TypeVar("_TFunc", bound=Callable[..., Any])
+
+# If `True`, we raise exceptions on use error (wrong parameter types, etc.).
+# If `False` we catch all errors and log a warning instead.
+_strict_mode = False
+
+_rerun_exception_ctx = threading.local()
+
+
+def check_strict_mode() -> bool:
```

Comment:
I don't love the name `check_strict_mode` - it sounds like it checks it and throws if it set, or something like that.

```suggestion
def is_strict_mode_on() -> bool:
```

Reply:
or just `strict_mode`